diff --git a/administrator/components/com_admin/admin.php b/administrator/components/com_admin/admin.php
index 2911a8d..3a7be9a 100644
--- a/administrator/components/com_admin/admin.php
+++ b/administrator/components/com_admin/admin.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 // No access check.
 
diff --git a/administrator/components/com_banners/banners.php b/administrator/components/com_banners/banners.php
index 3febd61..f232624 100644
--- a/administrator/components/com_banners/banners.php
+++ b/administrator/components/com_banners/banners.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_banners'))
 {
diff --git a/administrator/components/com_categories/categories.php b/administrator/components/com_categories/categories.php
index c540e37..eefdcbb 100644
--- a/administrator/components/com_categories/categories.php
+++ b/administrator/components/com_categories/categories.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 $input = JFactory::getApplication()->input;
 
diff --git a/administrator/components/com_config/config.php b/administrator/components/com_config/config.php
index dba688d..1605609 100644
--- a/administrator/components/com_config/config.php
+++ b/administrator/components/com_config/config.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 // Access checks are done internally because of different requirements for the two controllers.
 
diff --git a/administrator/components/com_contact/contact.php b/administrator/components/com_contact/contact.php
index e81a5fb..284f3b3 100644
--- a/administrator/components/com_contact/contact.php
+++ b/administrator/components/com_contact/contact.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_contact'))
 {
diff --git a/administrator/components/com_content/content.php b/administrator/components/com_content/content.php
index 2708fcf..8118e8e 100644
--- a/administrator/components/com_content/content.php
+++ b/administrator/components/com_content/content.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_content'))
 {
diff --git a/administrator/components/com_installer/installer.php b/administrator/components/com_installer/installer.php
index 2a4a882..1a572fa 100644
--- a/administrator/components/com_installer/installer.php
+++ b/administrator/components/com_installer/installer.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_installer'))
 {
diff --git a/administrator/components/com_languages/languages.php b/administrator/components/com_languages/languages.php
index 8abcd3f..e16e5ba 100644
--- a/administrator/components/com_languages/languages.php
+++ b/administrator/components/com_languages/languages.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_languages'))
 {
diff --git a/administrator/components/com_menus/menus.php b/administrator/components/com_menus/menus.php
index 48c763a..f6dc819 100644
--- a/administrator/components/com_menus/menus.php
+++ b/administrator/components/com_menus/menus.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_menus'))
 {
diff --git a/administrator/components/com_modules/modules.php b/administrator/components/com_modules/modules.php
index d3f5845..1d8f329 100644
--- a/administrator/components/com_modules/modules.php
+++ b/administrator/components/com_modules/modules.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_modules'))
 {
diff --git a/administrator/components/com_newsfeeds/newsfeeds.php b/administrator/components/com_newsfeeds/newsfeeds.php
index fb0184a..afeff23 100644
--- a/administrator/components/com_newsfeeds/newsfeeds.php
+++ b/administrator/components/com_newsfeeds/newsfeeds.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_newsfeeds'))
 {
diff --git a/administrator/components/com_plugins/plugins.php b/administrator/components/com_plugins/plugins.php
index 07d666e..34314c5 100644
--- a/administrator/components/com_plugins/plugins.php
+++ b/administrator/components/com_plugins/plugins.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_plugins'))
 {
diff --git a/administrator/components/com_tags/tags.php b/administrator/components/com_tags/tags.php
index d7503d9..32ee5e3 100644
--- a/administrator/components/com_tags/tags.php
+++ b/administrator/components/com_tags/tags.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 $input = JFactory::getApplication()->input;
 
diff --git a/administrator/components/com_templates/templates.php b/administrator/components/com_templates/templates.php
index eac3532..9911f97 100644
--- a/administrator/components/com_templates/templates.php
+++ b/administrator/components/com_templates/templates.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_templates'))
 {
diff --git a/administrator/components/com_users/users.php b/administrator/components/com_users/users.php
index d12cdf5..9b8484b 100644
--- a/administrator/components/com_users/users.php
+++ b/administrator/components/com_users/users.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_users'))
 {
diff --git a/administrator/components/com_weblinks/weblinks.php b/administrator/components/com_weblinks/weblinks.php
index eb8df8c..c77e32c 100644
--- a/administrator/components/com_weblinks/weblinks.php
+++ b/administrator/components/com_weblinks/weblinks.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 if (!JFactory::getUser()->authorise('core.manage', 'com_weblinks'))
 {
diff --git a/components/com_content/content.php b/components/com_content/content.php
index 0a5cff4..b67279a 100644
--- a/components/com_content/content.php
+++ b/components/com_content/content.php
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+JHtml::_('behavior.tabstate');
 
 require_once JPATH_COMPONENT.'/helpers/route.php';
 require_once JPATH_COMPONENT.'/helpers/query.php';
diff --git a/libraries/cms/html/behavior.php b/libraries/cms/html/behavior.php
index f8453e0..cb137ee 100644
--- a/libraries/cms/html/behavior.php
+++ b/libraries/cms/html/behavior.php
@@ -778,4 +778,29 @@ protected static function calendartranslation()
 			. ' Calendar._SMN = ' . json_encode($months_short) . ';'
 			. ' Calendar._TT = ' . json_encode($text) . ';';
 	}
+
+	/**
+	 * Add unobtrusive JavaScript support for a keeping tab state.
+	 *
+	 * Note that keeping tab state only work for inner tabs if the convension in accordance with the following example
+	 * parent tab = permissions
+	 * child tab = permission-<identifier>
+	 *
+	 * Each tab header "a" tag also should have a unique href attribute
+	 * 
+	 * @return  void
+	 *
+	 * @since   3.0
+	 */
+	public static function tabstate()
+	{
+		if (isset(self::$loaded[__METHOD__]))
+		{
+			return;
+		}
+		// Include jQuery
+		JHtml::_('jquery.framework');
+		JHtml::_('script', 'system/tabs-state.js', true, true);
+		self::$loaded[__METHOD__] = true;
+	}
 }
diff --git a/media/system/js/tabs-state.js b/media/system/js/tabs-state.js
new file mode 100644
index 0000000..749c125
--- /dev/null
+++ b/media/system/js/tabs-state.js
@@ -0,0 +1,42 @@
+/**
+ * @copyright   Copyright (C) 2005 - 2013 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+/**
+ * JavaScript behavior to allow selected tab to be remained after save or page reload
+ * keeping state in localstorage
+ */
+
+jQuery(function() {
+    
+    var loadTab = function() {
+        var $ = jQuery.noConflict();
+
+        jQuery(document).find('a[data-toggle="tab"]').on('click', function(e) {
+            // Store the selected tab href in localstorage
+            window.localStorage.setItem('tab-href', $(e.target).attr('href'));
+        });
+
+        var activateTab = function(href) {
+            var $el = $('a[data-toggle="tab"]a[href*=' + href + ']');
+            $el.tab('show');
+        }
+        if (localStorage.getItem('tab-href')) {
+            // Clean default tabs
+            $('a[data-toggle="tab"]').parent().removeClass('active');
+            var tabhref = localStorage.getItem('tab-href');
+            // Add active attribute for selected tab indicated by url
+            activateTab(tabhref);
+            // Check whether internal tab is selected (in format <tabname>-<id>)
+            var seperatorIndex = tabhref.indexOf('-');
+            if (seperatorIndex !== -1) {
+                var singular = tabhref.substring(0, seperatorIndex);
+                var plural = singular + "s";
+                activateTab(plural);
+            }
+        }
+    }
+    setTimeout(loadTab, 100);
+
+});
