Index: libraries/vendor/phpunit/phpunit/src/Framework/MockObject/Builder/ParametersMatch.php
===================================================================
--- libraries/vendor/phpunit/phpunit/src/Framework/MockObject/Builder/ParametersMatch.php	(date 1596445870909)
+++ libraries/vendor/phpunit/phpunit/src/Framework/MockObject/Builder/ParametersMatch.php	(date 1596445870909)
@@ -12,7 +12,7 @@
 /**
  * @internal This class is not covered by the backward compatibility promise for PHPUnit
  */
-interface ParametersMatch extends Match
+interface ParametersMatch extends Stub
 {
     /**
      * Sets the parameters to match for, each parameter to this function will
